modules/core/src/hal_replacement.hpp

```diff
@@ -214,6 +214,33 @@ inline int hal_ni_not8u(const uchar *src_data, size_t src_step, uchar *dst_data,
 #define cv_hal_not8u hal_ni_not8u
 //! @endcond
 
+/**
+Lookup table replacement
+Table consists of 256 elements of a size from 1 to 8 bytes having 1 channel or src_channels
+For 8s input type 128 is added to LUT index
+Destination should have the same element type and number of channels as lookup table elements
+@param src_data Source image data
+@param src_step Source image step
+@param src_channels Number of channels in source image
+@param lut_data Pointer to lookup table
+@param lut_channel_size Size of each channel in bytes
+@param lut_channels Number of channels in lookup table
+@param dst_data Destination data
+@param dst_step Destination step
+@param width Width of images
+@param height Height of images
+*/
+//! @addtogroup core_hal_interface_lut Lookup table
+//! @{
+inline int hal_ni_lut8u(const uchar *src_data, size_t src_step, size_t src_channels, const uchar* lut_data, size_t lut_channel_size, size_t lut_channels, uchar *dst_data, size_t dst_step, int width, int height) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
Many just introduced functions use 'int depth', 'int channels' (or a single 'type' parameter). Can we normalize it and at least follow this pattern for all functions? In this case it can be 'int src_depth', 'int lut_type'.

Reply:
Makes sense, will do this for next HAL functions